addons/snailmail/models/mail_thread.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo import api
+
+
+class MailThread(models.AbstractModel):
+    _inherit = 'mail.thread'
+
+    @api.model
+    def notify_cancel_by_type(self, notification_type):
+        if notification_type != 'snail':
+            return super().notify_cancel_by_type(notification_type)
```

Comment:
I think you need to call super in this case too, otherwise we'll miss the access check (and maybe other stuff). The method to be overridden should probably a sub method for only the type-specific code.

Also it feels like except from the access check itself, everything else is notification concern, and not thread concern?

Reply:
Yes you're right, we need for the access check but the implementation is different so we only need to execute the part of the code that check the access. I have added a optional parameter for that in the super method. It is not a very good solution, I know. Don't hesitate if you have a better idea. Thanks